src/test/java/org/mifos/integrationtest/cucumber/stepdef/BillPayStepDef.java

```diff
@@ -309,4 +323,195 @@ public void iCanExtractTheCallbackBodyAndAssertTheRtpStatus() {
         }
         assertThat(flag).isTrue();
     }
+
+    @Then("I should be able to extract response body from callback for biller unidentified")
+    public void iShouldBeAbleToExtractResponseBodyFromCallbackForBillerUnidentified() {
+        boolean flag = false;
+        List<ServeEvent> allServeEvents = getAllServeEvents();
+        for (int i = allServeEvents.size() - 1; i >= 0; i--) {
+            ServeEvent request = allServeEvents.get(i);
```

Comment:
Looks like this code is repeated all over, can we use a single method for finding in events and reuse them?

Reply:
is this not possible?

Reply:
i will mark this as a bug as it is still taking a little more lld than i thought

Reply:
https://fynarfin.atlassian.net/browse/PHEE-559